### PR TITLE
APF-420: New version available notice message added.

### PR DESCRIPTION
**Pull request by lcouget**
- Version update notice message added.

#### Additional details about this PR

### CODE REVIEW

File: Block/Adminhtml/System/Config/Fieldset/Payment.php [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            $this->getLocalModuleVersion(),
+            $this->getRemoteModuleVersion(),
+            '<'
+        );
+    }
+
+    /**
+     * @return string
+     */
+    protected function renderUpdateNotification(): string
```
Comment (dallin-beargroup) [inline Block/Adminhtml/System/Config/Fieldset/Payment.php:128]: An improvement could be here made by separating the HTML rendering to a separate **block class/view model** with .phtml template.
`$updateNotificationBlock = $this->layout->createBlock(<block_path>);
        return $customBlock->toHtml();`
where the template path is set in the constructor or as an overridden property in the block class.

File: Block/Adminhtml/System/Config/Fieldset/Payment.php [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+            ' - ' .
+            '<b>Latest Version: <span style="color:darkgreen">' . $this->getRemoteModuleVersion() . ')</span></b>. ' .
+            'See the <a target="_blank" href="' . self::MODULE_CHANGELOG_URL . '">ChangeLog</a>.' .
+            '</span></div></div>';
+        $html .='<br /><br />';
+
+        return $html;
+    }
+
+    protected function _getHeaderTitleHtml($element): string
```
Comment (dallin-beargroup) [inline Block/Adminhtml/System/Config/Fieldset/Payment.php:144]: Also could be a separate block for rendering the html

File: Model/Client.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+namespace Amazon\Pay\Model;
+
+use Magento\Framework\HTTP\Client\Curl;
+
+class Client extends Curl
+{
+    /**
+     * module JSON url
+     */
+    const URL = 'https://repo.packagist.org/p2/amzn/amazon-pay-magento-2-module.json';
```
Comment (dallin-beargroup) [inline Model/Client.php:12]: Maybe not an improvement, but a neat alternative. You can move this hard coded value to xml configuration.
`private string $repoSource;
public function __construct(string $repoSource = '')
{
    $this->repoSource = $repoSource;
}`
di.xml
`<type name="<class>">
    <arguments>
        <argument name="repoSource" xsi:type="string">https://repo.packagist.org/p2/amzn/amazon-pay-magento-2-module.json</argument>
    </arguments>
</type>`

**Review by dallin-beargroup — APPROVED** [review]

**Comment by lcouget** [comment]
Closing PC

**Pull request closed without merging**